### ISSUE TITLE
Running mpi code on the login node generate error log

### ISSUE THREAD

**Issue by cmd-ntrf** [ComputeCanada/puppet-magic_castle#12]
Originally reported by @cwant:
```
When I run an MPI code on our magic castle head node, I always get: [login01][[51019,1],0][btl_tcp_component.c:872:mca_btl_tcp_component_create_listen] bind() failed: Permission denied (13)
The code runs, but the message might be troublesome for users.
```

The message comes from SELinux denying the confined user to bind to a port. The easy but unrecommended solution is to temporarily deactivate SELinux : `sudo setenforce 0`.

We will need to investigate to determine if there is an SELinux parameter that would allow MPI to bind without error to a port.

**Comment by cmd-ntrf** [comment]
I used audit2allow and setroubleshoot-server to pinpoint the required sebool to remove the error.
```
$ sudo audit2allow -w -a
type=AVC msg=audit(1569425734.271:470): avc:  denied  { name_bind } for  pid=4596 comm="main" src=1024 scontext=user_u:user_r:user_t:s0 tcontext=system_u:object_r:unreserved_port_t:s0 tclass=tcp_socket permissive=0
[...]
    Allow access by executing:
    # setsebool -P selinuxuser_tcp_server 1
```

We will add the boolean activation in the base puppet module:
```
selinux::boolean { 'selinuxuser_tcp_server': }
```

**Comment by cmd-ntrf** [comment]
Fixed by df4074f.
